### ISSUE TITLE
Kubernetes support

### ISSUE THREAD

**Issue by zacanger** [entropic-dev/entropic#116]
Might be nice to have a helm chart or k8s files rather than using `docker-compose` and/or manually starting, for deployment. I could work on this if you think it'd be valuable.

* Fully dockerized local dev setup: #119
* k8s setup for easy prod standup: #204

**Comment by ceejbot** [comment]
I would *love*  this. We need easy start-up both for development and for easy stand-up by  administrators.

**Comment by toddself** [comment]
@zacanger i'd love to help out with this if you need any assistance!

**Comment by zacanger** [comment]
Definitely! I'm still learning Kubernetes so any help is appreciated. I got a rough start [here](https://github.com/zacanger/entropic/commit/23309c90c3554ac26ca77e389d23c4ac5453ca60) but took a break from it due to minikube using literally all of my laptop's resources.

**Comment by toddself** [comment]
awesome i can give this some testing!

also we might consider using something like [pulumi](https://www.pulumi.com/kubernetes/) or another tool rather than having to manage yaml by hand. i ... dislike yaml a lot.

**Comment by zacanger** [comment]
Some tool to enable folks to work with anything other than yaml would be amazing, I think. I spend way too much time fiddling with indentation. Even just using a js to yaml package might be a pretty good solution. @ceejbot what do you think?

**Comment by zacanger** [comment]
@toddself I made some pretty big changes over here https://github.com/zacanger/entropic/tree/feature/k8s/misc/k8s -- now building from plain JS objects, and spent some time cleaning it up.

**Comment by toddself** [comment]
oh awesome! i've been heads down on my day job (and writing a Package.toml RFC) but will check this out tonight!

**Comment by krainboltgreene** [comment]
I know quite a bit about this surface of application development and I would love to do a pair review to transfer the knowledge.

**Comment by alexindigo** [comment]
@zacanger is there anything left to tackle? I spent last year and half doing exactly this, so I'd like to chip in. :)

**Comment by zacanger** [comment]
@alexindigo thanks! I think #204 is good to go, but definitely welcome more eyes on it!

**Comment by alexindigo** [comment]
Thanks @zacanger, I see it's k8s related changes. Are you interested in local docker development?

**Comment by alexindigo** [comment]
And you didn't go with helm just because of yaml or didn't want to have go templates in the stack?

**Comment by zacanger** [comment]
Local docker env was set up in #119, so now running locally should be as simple as `npm i` and `docker-compose up`.

I didn't go with Helm mostly because I wanted to avoid the Helm 2 -> Helm 3 churn. I think Helm would make a lot of sense, but maybe not until Helm 3 is stable

**Comment by alexindigo** [comment]
Thanks. For me, I had bad time with docker default networking, so I ended up forcing bridge network and also starting via `docker-compose run...` so container would be destroyed on exit, but for multiple containers it's more pain than gain :)

**Comment by alexindigo** [comment]
Btw, looking at both changesets, looks like some kind of shared configuration will be very useful,
some default config file with initial/dev values and then support for overriding it with env vars, which could be injected via k8s manifest at deploy time. So we can have baked (stable) docker image with specific code version and run it against different environments without rebuilding the world.

I see that you're already going this direction, but it seems to be still spread over multiple places.
Curious if you gave it deep thought and current state is what you want or it's just incremental steps towards something more "sophisticated".

**Comment by zacanger** [comment]
I haven't given that much thought in these two PRs, but think it would definitely be valuable to have less scattered and duplicated config around. Definitely open to suggestions! That might make a good topic for discussion on [Discourse](http://discourse.entropic.dev/)

**Comment by zacanger** [comment]
Closed in 63731622bf9e6d98de25fad0b8f08b5db8d8e7dd
